functions/date-time-offset/1.2/function.json

```diff
@@ -0,0 +1,128 @@
+{
+  "description": "Calculate a date, time or date with time based on a provided date in text (YYYY-MM-DD hh:mm:ss) or the current date",
+  "label": "Date/Time Offset",
+  "category": "Misc",
+  "icon": {
+    "name": "DateTimeIcon",
+    "color": "Yellow"
+  },
+  "options": [
+    {
+      "name": "customStartDate",
+      "label": "Specify your date in YYYY-MM-DD hh:mm:ss notation",
+      "meta": {
+        "type": "Text"
+      }
+    },
+    {
+      "name": "currentDate",
+      "label": "OR use the current date/time ",
+      "meta": {
+        "type": "Boolean"
+      }
+    },
+    {
+      "name": "businessDays",
+      "label": "Business days only",
+      "meta": {
+        "type": "Boolean"
+      },
+      "info": "If checked, the calculation will skip weekends (Saturday and Sunday) and will only count working days (Monday to Friday). This option only works in combination with offset type \"Days\"."
+    },
+    {
+      "name": "offsetType",
+      "label": "Offset Type",
+      "meta": {
+        "type": "Select",
+        "validations": {
+          "required": true
+        },
+        "values": [
+          { "label": "Seconds", "value": "ss" },
+          { "label": "Minutes", "value": "mm" },
+          { "label": "Hours", "value": "hh" },
+          { "label": "Days", "value": "DD" },
+          { "label": "Weeks", "value": "WW" },
+          { "label": "Months", "value": "MM" },
+          { "label": "Years", "value": "YYYY" }
+        ]
+      }
+    },
+    {
+      "name": "offset",
+      "label": "offset",
+      "meta": {
+        "type": "Number",
+        "validations": {
+          "required": true
+        }
+      }
+    },
+    {
+      "name": "resultType",
+      "label": "Result type",
+      "meta": {
+        "type": "Select",
+        "validations": {
+          "required": true
+        },
+        "values": [
+          { "label": "Date", "value": "D" },
+          { "label": "Date Time", "value": "DT" },
+          { "label": "Time", "value": "T" },
+          { "label": "Unix Timestamp (epoch)", "value": "UT" }
```

Comment:
does this return a string or an integer?

Reply:
It returns a string value because the `format()` method of "date-fns" is used and always returns a string. Nevertheless, this value can still be used in a number property assignment, the value can be used in a condition step (e.g. "1738756275" is greater than "1738756249"), and we can also calculate the difference using the Expression step. I could also enforce a number type value by using the `Number()` method. What do you think?

Reply:
I've now enforced the value of a unix timestamp to be a number.